### PR TITLE
Alternative RestSharp v109 update simplefactory

### PR DESCRIPTION
**Pull request by timothodge**
Given issues with updating to v107 we could attempt to go up to v109 and leverage a simple factory that will hold many instances based on baseUrl. See attached link.

https://restsharp.dev/usage.html#simple-factory

This is a possible path if we don't have a method forward with v107 changes.

### CODE REVIEW

**Review by lifebeyondfife — DISMISSED** [review]
Before moving outside of beta, we should remove all the debugging statements and reinsert the mock tests. Moving to a fully async/await model will likely be needed for this. We can discuss as a team whether this goes to general availability as a minor or major version increase.

File: smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs [hunk 1]
```diff
@@ -107,10 +116,17 @@ public DefaultHttpClient(RestClient httpClient, JsonSerializer jsonSerializer)
...
-            this.httpClient.FollowRedirects = true;
+            //this.httpClient.FollowRedirects = true;
             this.jsonSerializer = jsonSerializer;
         }
 
+
+        public virtual HttpResponse Request(HttpRequest smartsheetRequest, string objectType, string file, string fileType) {
+            var task = this.RequestAsync(smartsheetRequest, objectType, file, fileType);
+            task.Wait(); 
+            var result = task.Result;
```
Comment (rzaracota) [inline smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs]: is this local variable redundant or does the result getter have a side-effect? :sweat_smile:

File: smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs [hunk 2]
```diff
@@ -144,20 +160,28 @@ public virtual HttpResponse Request(HttpRequest smartsheetRequest, string object
             restRequest.AddFile("file", File.ReadAllBytes(file), new FileInfo(file).Name, fileType);
             if (smartsheetRequest.Entity != null && smartsheetRequest.Entity.GetContent() != null)
             {
-                restRequest.AddParameter(objectType.ToLower(), System.Text.Encoding.Default.GetString(smartsheetRequest.Entity.Content),
-                    smartsheetRequest.Entity.ContentType, ParameterType.RequestBody);
+                //restRequest.AddParameter(objectType.ToLower(), System.Text.Encoding.Default.GetString(smartsheetRequest.Entity.Content),
```
Comment (rzaracota) [inline smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs]: should we move these commented out point to a different mr and just remove them for the beta?

File: smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs [hunk 3]
```diff
@@ -223,28 +257,70 @@ public virtual HttpResponse Request(HttpRequest smartsheetRequest)
...
+                        restRequest = restRequest.AddBody(param);
+                    }
+
                 }
 
                 // Set the client base Url.
-                httpClient.BaseUrl = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
-
+                //httpClient.BaseUrl = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
+                Uri baseUrlAsURI = new Uri(smartsheetRequest.Uri.GetLeftPart(UriPartial.Authority));
```
Comment (rzaracota) [inline smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs]: it looks like these 3ish lines are repeated in several places. it might be helpful to extract a method or function

File: smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs [hunk 4]
```diff
@@ -223,28 +257,70 @@ public virtual HttpResponse Request(HttpRequest smartsheetRequest)
...
+                restResponse = restResponseAsTask.Result;
                 timer.Stop();
 
                 LogRequest(restRequest, restResponse, timer.ElapsedMilliseconds);
 
                 if (restResponse.ResponseStatus == ResponseStatus.Error)
                 {
-                    throw new HttpClientException("There was an issue connecting.");
+                    Boolean restClientDebugging = false;
+                    if (restClientDebugging) {
```
Comment (rzaracota) [inline smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs:286]: nitpick: this will work for now but a possible alternative would be conditional compilation directives if they make sense in this context: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation

File: smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs [hunk 5]
```diff
@@ -223,28 +257,70 @@ public virtual HttpResponse Request(HttpRequest smartsheetRequest)
...
+                        }
+
+                        builder.Append(" There was an issue connecting.");
+                        builder.Append(" RestResponse code is " + restResponse.StatusCode.ToString() + "\n");
+                        builder.Append(" RestResponse ErrorMessage is " + restResponse.ErrorMessage + "\n");
+                        builder.Append(" RestResponse Content is " + restResponse.Content + "\n");
+
+                    throw new HttpClientException(builder.ToString());
+                    } else {
+                        throw new HttpClientException("There was an issue connecting.");
```
Comment (rzaracota) [inline smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs]: should this include the error message if there is one?

Comment (timothodge) [reply smartsheet-csharp-sdk/main/Smartsheet/Api/Internal/Http/DefaultHttpClient.cs]: Have added one now 👍

**Review by rzaracota — APPROVED** [review]
